### ISSUE TITLE
Imported oci_core_volume_group does not contain all details; triggers replace on next apply

### ISSUE THREAD

**Issue by jeliker** [terraform-providers/terraform-provider-oci#1042]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version
Terraform v0.12.21
+ provider.null v2.1.2
+ provider.oci v3.71.0
+ provider.template v2.1.2

### Affected Resource(s)
oci_core_volume_group

### Terraform Configuration Files
```hcl
resource "oci_core_instance" "the_instance" {
 . . . and so on
}
resource "oci_core_volume" "the_storage" {
  count = length(var.storage_details)
  . . . and so on
}

resource "oci_core_volume_group" "the_volume_group" {
  #Required
  availability_domain = var.availability_domain
  compartment_id      = var.compartment_id
  
  source_details {
    type       = "volumeIds"
    volume_ids = concat([oci_core_instance.the_instance.boot_volume_id], oci_core_volume.the_storage[*].id)
  }
}
```

### Debug Output
When I import the existing  ```oci_core_volume_group``` I see this in state:

```hcl
{
  "module": "module.inst1_dev",
  "mode": "managed",
  "type": "oci_core_volume_group",
  "name": "the_volume_group",
  "provider": "provider.oci",
  "instances": [
	{
	  "schema_version": 0,
	  "attributes": {
		"availability_domain": "OQhv:US-ASHBURN-AD-3",
		"compartment_id": "ocid1.compartment.oc1..the.ocid",
		"defined_tags": {},
		"display_name": "Inst1_Dev",
		"freeform_tags": {},
		"id": "ocid1.volumegroup.oc1.iad.the.ocid",
		"is_hydrated": null,
		"size_in_gbs": "150",
		"size_in_mbs": "153600",
		"source_details": [
		  {
			"type": "volumeIds",
			"volume_group_backup_id": "",
			"volume_group_id": "",
			"volume_ids": [
			  "ocid1.bootvolume.oc1.iad.the.ocid"
			]
		  }
		],
		"state": "AVAILABLE",
		"time_created": "2020-03-07 03:35:00.845 +0000 UTC",
		"timeouts": {
		  "create": null,
		  "delete": null,
		  "update": null
		},
		"volume_ids": [
		  "ocid1.bootvolume.oc1.iad.the.ocid",
		  "ocid1.volume.oc1.iad.the.ocid",
		  "ocid1.volume.oc1.iad.the.ocid"
		]
	  },
	  "private": "gibberishgibberishgibberish"
	}
  ]
}
```

and when I create new I see this in state

```hcl
{
  "module": "module.inst1_dev",
  "mode": "managed",
  "type": "oci_core_volume_group",
  "name": "the_volume_group",
  "provider": "provider.oci",
  "instances": [
	{
	  "schema_version": 0,
	  "attributes": {
		"availability_domain": "OQhv:US-ASHBURN-AD-3",
		"compartment_id": "ocid1.compartment.oc1..the.ocid",
		"defined_tags": {},
		"display_name": "Inst1_Dev",
		"freeform_tags": {},
		"id": "ocid1.volumegroup.oc1.iad.the.ocid",
		"is_hydrated": null,
		"size_in_gbs": "150",
		"size_in_mbs": "153600",
		"source_details": [
		  {
			"type": "volumeIds",
			"volume_group_backup_id": "",
			"volume_group_id": "",
			"volume_ids": [
			  "ocid1.bootvolume.oc1.iad.the.ocid",
			  "ocid1.volume.oc1.iad.the.ocid",
			  "ocid1.volume.oc1.iad.the.ocid"
			]
		  }
		],
		"state": "AVAILABLE",
		"time_created": "2020-04-22 19:00:20.726 +0000 UTC",
		"timeouts": null,
		"volume_ids": [
		  "ocid1.bootvolume.oc1.iad.the.ocid",
		  "ocid1.volume.oc1.iad.the.ocid",
		  "ocid1.volume.oc1.iad.the.ocid"
		]
	  },
	  "private": "gibberishgibberishgibberish",
	  "dependencies": [
		"module.compartment_nonprod_app.oci_identity_compartment.the_compartment",
		"module.inst1_dev.oci_core_instance.the_instance",
		"module.inst1_dev.oci_core_volume.the_storage",
		"module.vcn_main_nsg_inst1_dev.oci_core_network_security_group.the_nsg",
		"module.vcn_main_subnet_intapps.oci_core_subnet.the_subnet"
	  ]
	}
  ]
}
```

Notice above that imported ```oci_core_volume_group``` does not have all ```volume_ids``` under ```source_details``` (even though it does have proper quantity and values under top-level ```volume_ids```)

### Expected Behavior
Imported ```oci_core_volume_group``` should contain all ```volume_ids``` as identified in existing Volume Group

### Actual Behavior
Imported ```oci_core_volume_group``` does not contain all ```volume_ids``` as identified in existing Volume Group

### Steps to Reproduce

1. Create ```oci_core_volume_group``` and note new OCID
2. Remove from state with ```terraform state rm _resource_id_```
3. Import ```terraform import _resource_id_ ocid.somethingsomething```
4. Check with ```terraform plan``` and note prompt to "replace" because of missing ```volume_ids``` detail.

**Comment by varmax2511** [comment]
Hi @jeliker -- thanks for reporting this. While we investigate I would like to understand if subsequent apply causes an in-place update on the resource for you or Create/Destroy?

**Comment by varmax2511** [comment]
Hi @jeliker -- thanks for reporting this. While we investigate, can you try the suggestion here https://github.com/terraform-providers/terraform-provider-oci/issues/1037#issuecomment-617257184 and let us know if it unblocks you?

**Comment by jeliker** [comment]
I tried manually adding ```volume_ids``` to state file but apply still forced a "replace."

I am not blocked necessarily since recreating the volume group is tolerable in my current state. However, I hope you will still consider the bug report here as I do not believe what I am seeing is intended functionality.

**Comment by varmax2511** [comment]
Hi @jeliker -- thank you for your patience. We are unable to reproduce the issue per the steps and TF config listed. After creating the resource, removing it from state and then importing it back,both `volume_ids` fields are consistent.

```
resource "oci_core_volume_group" "the_volume_group" {
  #Required
  availability_domain = var.availability_domain
  compartment_id      = var.compartment_id
  
  source_details {
    type       = "volumeIds"
    volume_ids = concat([oci_core_instance.the_instance.boot_volume_id], oci_core_volume.the_storage[*].id)
  }
}
```
Can you please try reproducing this issue again at your end? Please share the Terraform config that is consistently able to reproduce the issue for us to assist you better.

**Comment by jeliker** [comment]
I encountered the same issue at least 14 times in my production scenario before opening this issue. Now I tried a very specific test case and do not see the issue. I repeated one of my production issues and find it's working.  I've tried three different times/different ways and it works in all cases now.

I wonder if there was a backend-api update in recent days that has "fixed" the issue? I also experienced a similar "used-to-fail" issues related to changing ```hostname_label``` on a ```oci_core_instance``` instance (i.e. it used to not work then it worked but only changing vnic value and not top-level value now it works 100%).

In any case, it seems the issue is gone. That is, I am unable to reproduce it today (having experienced it, as I say, for the past two weeks).
